t/apicast-policy-rate-limit.t

```diff
@@ -0,0 +1,995 @@
+                      rate = 20,
+                      burst = 10
+                    },
+                    {
+                      name = "connections",
+                      key = "test6_2",
+                      conn = 20,
+                      burst = 10,
+                      delay = 0.5
+                    },
+                    {
+                      name = "fixed_window",
+                      key = "test6_3",
+                      count = 20,
+                      window = 10
+                    }
+                  },
+                  redis_url = "redis://localhost:6379/1"
+                }
+              }
+            }
+          }
+        }
+      }
+    })
+  }
+  lua_shared_dict limiter 1m;
+
+--- config
+  include $TEST_NGINX_APICAST_CONFIG;
+
+  location /flush_redis {
+    content_by_lua_block {
+      local redis = require('resty.redis'):new()
+      redis:connect('127.0.0.1', 6379)
+      redis:select(1)
+      redis:del("test6_1", "test6_2", "test6_3")
+    }
+  }
+
+--- pipelined_requests eval
+["GET /flush_redis","GET /"]
+--- error_code eval
+[200, 200]
+--- no_error_log
+[error]
+need to delay by
+
+=== TEST 7: Rejected (conn).
+Return 429 code.
+--- http_config
+  include $TEST_NGINX_UPSTREAM_CONFIG;
+  lua_package_path "$TEST_NGINX_LUA_PATH";
+
+  init_by_lua_block {
+    require "resty.core"
+    ngx.shared.limiter:flush_all()
+    require('apicast.configuration_loader').mock({
+      services = {
+        {
+          id = 42,
+          proxy = {
+            policy_chain = {
+              {
+                name = "apicast.policy.rate_limit",
+                configuration = {
+                  limiters = {
+                    {
+                      name = "connections",
+                      key = "test7",
+                      conn = 1,
+                      burst = 0,
+                      delay = 2
+                    }
+                  },
+                  redis_url = "redis://localhost:6379/1"
+                }
+              },
+              {
+                name = "apicast.policy.rate_limit",
+                configuration = {
+                  limiters = {
+                    {
+                      name = "connections",
+                      key = "test7",
+                      conn = 1,
+                      burst = 0,
+                      delay = 2
+                    }
+                  },
+                  redis_url = "redis://localhost:6379/1"
+                }
+              }
+            }
+          }
+        }
+      }
+    })
+  }
+  lua_shared_dict limiter 1m;
+
+--- config
+  include $TEST_NGINX_APICAST_CONFIG;
+
+  location /flush_redis {
+    content_by_lua_block {
+      local redis = require('resty.redis'):new()
+      redis:connect('127.0.0.1', 6379)
+      redis:select(1)
+      redis:del("test7")
+    }
+  }
+
+--- pipelined_requests eval
+["GET /flush_redis","GET /"]
+--- error_code eval
+[200, 429]
```

Comment:
I'm not sure about this one. `conn` is set to 1 in the config. This means that it will only allow 1 concurrent connection.
So if I understood this correctly, the first `GET /` should be a 200 for sure. Even if we added more, they could be 200s as well as long as there isn't more than 1 concurrent connection.

Reply:
@davidor I define the policy twice here, so we can open 2 connections by "GET /", and the first "GET /" is a 429.